velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1266,6 +1273,26 @@ struct FormatDateTimeFunction {
     }
     return true;
   }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<TimestampWithTimezone>& timestampWithTimezone,
+      const arg_type<Varchar>& formatString) {
+    ensureFormatter(formatString);
+
+    const auto milliseconds = *timestampWithTimezone.template at<0>();
+    Timestamp timestamp = Timestamp::fromMillis(milliseconds);
+    int16_t timeZoneId = *timestampWithTimezone.template at<1>();
+    auto* timezonePtr = date::locate_zone(util::getTimeZoneName(timeZoneId));
+
+    auto formattedResult = jodaDateTime_->format(timestamp, timezonePtr);
+    auto resultSize = formattedResult.size();
+    result.resize(resultSize);
+    if (resultSize != 0) {
+      std::memcpy(result.data(), formattedResult.data(), resultSize);
```

Comment:
why not `result.setNoCopy(formattedResult)` ? In that case, I think you don't need result.resize as well. cc: @laithsakka

Reply:
`setNoCopy()` only works if you have StringView. Here, date time formatter returns a std::string, so there is no way around it. I put a TODO above though, for us to remember to add an API to the formatter to allow this to be done without the extra copy.